src/Extractor/HttpClient.php

```diff
@@ -198,8 +198,8 @@ public function fetch(UriInterface $url, bool $skipTypeVerification = false, arr
         }
 
         // be sure to remove ALL other conditional comments for IE
-        // (regex found here: https://stackoverflow.com/a/137831/569101)
-        preg_match_all('/<!--\[if\s(?:[^<]+|<(?!!\[endif\]-->))*<!\[endif\]-->/mi', $body, $matchesConditional);
+        // (regex inspired from here: https://stackoverflow.com/a/55083809/954513)
+        preg_match_all('/<!--(?:\[| ?<!)(?:[^-]|-[^-]|--+[^->])*--+>/mi', $body, $matchesConditional);
```

Comment:
The regex looks mostly correct to me:

- The left branch in the first non-capturing group will match [downlevel-hidden conditional comments](https://en.wikipedia.org/wiki/Conditional_comment#Downlevel-hidden_conditional_comment) and the rest of the regex will match remainder of the conditional. It will also match just the opening part of the [downlevel-revealed tag](https://en.wikipedia.org/wiki/Conditional_comment#Downlevel-revealed_conditional_comment) encapsulated within a comment.
- The right branch will then match the closing part of the downlevel-revealed tag (encapsulated within a comment).
- Comments [cannot really contain `--` so](https://developer.mozilla.org/en-US/docs/Web/API/Comment) the `--+[^->]` branch is redundant.

It will not handle standalone downlevel-revealed tags but those were not handled before either and are probably rare since they are not valid HTML syntax.

Reply:
Removing `|--+[^->` actually breaks the original case I am fixing as it doesn't match the `if [!mso]` line

![2023-08-13-195136_736x279](https://github.com/j0k3r/graby/assets/226063/f669c1e6-9313-4fc2-b549-8991ff9e9ed7)

Reply:
Ah. That is actually not a valid HTML – it should be `<!--[if !mso]><!-->`. But I guess if real sites use this, we need to support this abomination as well. Maybe add a comment that that branch is to support non-compliant

Reply:
Actually, looks like [HTML 5 now allows double dashes in comments](https://github.com/whatwg/html/issues/1351) so maybe just replace `(?:[^-]|-[^-]|--+[^->])*--+>` with a non-greedy match like:

```suggestion
        preg_match_all('/<!--(?:\[| ?<!).+?-->/mi', $body, $matchesConditional);
```

Reply:
Your pattern does not match all things matched by mine:
![2023-09-12-210344_717x420](https://github.com/j0k3r/graby/assets/226063/a94b2b30-9a76-4e5e-868f-5a81cdfa6eb0)

Reply:
Oh, right, it will also need `dotall` (s) flag to have `.` match line breaks.

Reply:
It looks like to work indeed 8c225bd05edac28ca323070b78a68ccca9afbc76